src/org/omegat/core/data/ExternalTMXEntry.java

```diff
@@ -0,0 +1,125 @@
+/**************************************************************************
+ OmegaT - Computer Assisted Translation (CAT) tool
+          with fuzzy matching, translation memory, keyword search,
+          glossaries, and translation leveraging into updated projects.
+
+ Copyright (C) 2010 Alex Buloichik
+               2012 Guido Leenders, Thomas Cordonnier
+               2013 Aaron Madlon-Kay
+               2014 Alex Buloichik, Aaron Madlon-Kay
+               Home page: http://www.omegat.org/
+               Support center: https://omegat.org/support
+
+ This file is part of OmegaT.
+
+ OmegaT is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ OmegaT is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ **************************************************************************/
+
+package org.omegat.core.data;
+
+import java.util.List;
+import java.util.Objects;
+
+import org.omegat.util.TMXProp;
+
+/**
+ * Storage for TMX entry.
+ *
+ * Variables in this class can be changed only before store to ProjectTMX. After that, all values must be
+ * unchangeable.
+ *
+ * Only RealProject can create and change TMXEntry objects.
+ *
+ * @author Alex Buloichik (alex73mail@gmail.com)
+ * @author Guido Leenders
+ * @author Aaron Madlon-Kay
+ */
+public class ExternalTMXEntry extends TMXEntry {
+    public List<TMXProp> otherProperties;
```

Comment:
I think this can be and should be private.

Reply:
I did it like that only because all other fields remained public in the other classes. And setting other fields as private requires lot of changes in the code. But I agree that it can be discutable. Tell me what you think

Reply:
My idea is...
 As we discussed in developer mailing list, List<> is flexible but not optimal for memory consumption.
This is only chance to change access modifier for `otherProperties` because `ExternalTMXEntry` is new class. Other fields should public for backward compatibility.
 After we introduce `private otherProperties` and access methods, we can change it into other representation.
If we put it as `public`, we lost a chance to change it in future.